src/probestack.rs

```diff
@@ -43,18 +43,32 @@
 
 #![cfg(not(windows))] // Windows already has builtins to do this
 
+extern "C" {
+    pub fn __rust_probestack();
+}
+
 #[naked]
 #[no_mangle]
 #[cfg(all(target_arch = "x86_64", not(feature = "mangled-names")))]
-pub unsafe extern "C" fn __rust_probestack() {
+pub unsafe extern "C" fn __rust_probestack_wrapper() {
     // Our goal here is to touch each page between %rsp+8 and %rsp+8-%rax,
     // ensuring that if any pages are unmapped we'll make a page fault.
     //
     // The ABI here is that the stack frame size is located in `%eax`. Upon
     // return we're not supposed to modify `%esp` or `%eax`.
     asm!("
+        .cfi_endproc
```

Comment:
I think the endproc/pushsection/popsection/startproc dance merits a comment explaining why endproc/startproc are needed.